File: MediafyDemoAppObjectiveC/MediafyDemoAppObjectiveC/IntegrationCases/SDK/MediafyBannerAdViewController.m
/*
 * Mediafy SDK
 *
 * Copyright 2024 Mars Technologies Ltd. All rights reserved.
 */

#import "MediafyBannerAdViewController.h"

@interface MediafyBannerAdViewController ()

@property (nonatomic) MediafyAdView * adView;

@end

@implementation MediafyBannerAdViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    [self loadAd];
}

- (void)loadAd {
    // 1. Create a MediafyAdView
    self.adView = [[MediafyAdView alloc] initWithFrame:CGRectMake(0, 0, 320, 50)];
    
    // 2. Configure the MediafyAdView
    self.adView.delegate = self;
    [self.adView setAdSizes:@[[NSValue valueWithCGSize:CGSizeMake(320, 50)]]];
    
    // Add Mediafy ad view to the app UI
    [self.containerAdView addSubview:self.adView];
    
    // 3. Load the ad
    [self.adView loadAd];
}

// MARK: - MediafyAdViewDelegate

- (UIViewController *)adViewPresentationController {
    return self;
}

- (void)adView:(MediafyAdView *)adView didFailToReceiveAdWith:(NSError *)error {
    // Called when Mediafy SDK failed to load ad
    NSLog(@"Did fail to receive ad with error: %@", error.localizedDescription);
}

- (void)adView:(MediafyAdView *)adView didReceiveAdWithAdSize:(CGSize)adSize {
    // Called when ad is loaded
    
    // Update the constraints of the container view
    self.containerWidthConstraint.constant = adSize.width;
    self.containerHeightConstraint.constant = adSize.height;
}

- (void)adViewWillPresentModal:(MediafyAdView *)adView {
    // Called when modal is about to be presented
}

- (void)adViewDidDismissModal:(MediafyAdView *)adView {
    // Called when modal is dismissed
}

- (void)adViewWillLeaveApplication:(MediafyAdView *)adView {
    // Called when the application is about to enter the background
}

@end

File: MediafyDemoAppSwift/MediafyDemoAppSwift/IntegrationCases/SDK/MediafyBannerAdViewController.swift
/*
 * Mediafy SDK
 *
 * Copyright 2024 Mars Technologies Ltd. All rights reserved.
 */

import UIKit
import MediafySDK

final class MediafyBannerAdViewController: AdBaseViewController {
    
    private var adView: MediafyAdView!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        loadAd()
    }
    
    private func loadAd() {
        // 1. Create a MediafyAdView
        adView = MediafyAdView(
            frame: CGRect(origin: .zero, size: CGSize(width: 320, height: 50))
        )
        
        // 2. Configure the MediafyAdView
        adView.delegate = self
        adView.adSizes = [CGSize(width: 320, height: 50)]
        
        // Add the ad view to the app UI
        containerAdView.addSubview(adView)
        
        // 3. Load the ad
        adView.loadAd()
    }
}

// MARK: - MediafyAdViewDelegate

extension MediafyBannerAdViewController: MediafyAdViewDelegate {
    
    func adViewPresentationController() -> UIViewController? {
        self
    }
        
    func adView(_ adView: MediafyAdView, didFailToReceiveAdWith error: any Error) {
        print("Did fail to receive ad with error: \(error.localizedDescription)")
    }

    func adView(_ adView: MediafyAdView, didReceiveAdWithAdSize adSize: CGSize) {
        // Called when ad is loaded
        
        // Update the constraints of the container view
        containerWidthConstraint.constant = adSize.width
        containerHeightConstraint.constant = adSize.height
    }
    
    func adViewWillPresentModal(_ adView: MediafyAdView) {
        // Called when modal is about to be presented
    }
    
    func adViewDidDismissModal(_ adView: MediafyAdView) {
        // Called when modal is dismissed
    }

    func adViewWillLeaveApplication(_ adView: MediafyAdView) {
        // Called when the application is about to enter the background
    }
}
